R/make_clean_names.R

```diff
@@ -24,58 +51,91 @@
 #' x
 #' names(x) <- make_clean_names(names(x))
 #' x # now has cleaned names
-
+#'
 #' # if you prefer camelCase variable names:
 #' make_clean_names(names(x), "small_camel")
 #'
 #' # similar to janitor::clean_names(poorly_named_df):
 #' # not run:
 #' # make_clean_names(names(poorly_named_df))
-#' 
-make_clean_names <- function(string, case = c(
-  "snake", "lower_camel", "upper_camel", "screaming_snake",
-  "lower_upper", "upper_lower", "all_caps", "small_camel",
-  "big_camel", "old_janitor", "parsed", "mixed", "none"
-)) {
+#'
+#' @importFrom stringi stri_trans_general
+#' @importFrom stringr str_replace str_replace_all
+make_clean_names <- function(string,
+                             case = c(
+                               "snake", "lower_camel", "upper_camel", "screaming_snake",
+                               "lower_upper", "upper_lower", "all_caps", "small_camel",
+                               "big_camel", "old_janitor", "parsed", "mixed", "none"
+                             ),
+                             replace=
+                               c(
+                                 "'"="",
+                                 "\""="",
+                                 "%"=".percent_",
+                                 "#"=".number_"
+                               ),
+                             ascii=TRUE) {
   
-  # old behavior, to provide easy fix for people whose code breaks with the snakecase integration
+  # Handling "old_janitor" case for backward compatibility
   case <- match.arg(case)
   if (case == "old_janitor") {
     return(old_make_clean_names(string))
   }
-  
-  ### new behaviour with snakecase integration
-  # Takes a data.frame, returns the same data frame with cleaned names
-  old_names <- string
-  new_names <- old_names %>%
-    gsub("'", "", .) %>% # remove single quotation marks
-    gsub("\"", "", .) %>% # remove double quotation marks
-    gsub("%", ".percent_", .) %>% # starting with "." as a workaround, to make
-    # ".percent" a valid name. The "." will be replaced in the call to to_any_case
-    # via the preprocess argument anyway.
-    gsub("#", ".number_", .) %>%
-    gsub("^[[:space:][:punct:]]+", "", .) %>% # remove leading spaces & punctuation
-    make.names(.) %>%
-    # Handle dots, multiple underscores, case conversion, string transliteration
-    # Parsing option 4 removes underscores around numbers, #153
-    snakecase::to_any_case(.,
-      case = case, sep_in = "\\.",
-      transliterations = c("Latin-ASCII"), parsing_option = 1,
+
+  replaced_names <-
+    stringr::str_replace_all(
+      str=string,
+      pattern=replace
+    )
+  transliterated_names <-
+    if (ascii) {
```

Comment:
This is all so readable :heart_decoration:

Reply:
:)